src/retasc/validator/models.py

```diff
@@ -0,0 +1,37 @@
+from pydantic import BaseModel, Field
+
+
+class JiraIssue(BaseModel):
+    """Represents a Jira issue, which can have subtasks."""
+
+    template: str = Field(description="The template string for the jira issue.")
+    subtasks: list["JiraIssue"] = Field(
+        default=[], description="The subtasks for the jira issue."
+    )
+
+
+class Prerequisites(BaseModel):
+    """Defines the prerequisites needed for a rule."""
+
+    pp_schedule_item_name: str = Field(description="The name of the pp schedule item.")
+    days_before_or_after: int = Field(
+        description=(
+            "The number of days to adjust the schedule relative to the PP schedule item date. "
+            "A negative value indicates the number of days before the PP schedule item date, "
+            "while a positive value indicates the number of days after the PP schedule item date. "
+            "This value helps determine the target date for creating Jira issues."
+        ),
+    )
+    dependent_rules: list[str] = Field(
+        default=[],
+        description="The dependent rules for the prerequisite schedule item.",
+    )
+
+
+class Rule(BaseModel):
+    """Represents a rule which includes prerequisites and Jira issues."""
+
+    version: int = Field(description="The version of the rule.")
```

Comment:
This should indicate schema version:

```suggestion
    version: int = Field(description="The version of the rule schema. The latest version is 1.")
```

We will need to think about how to keep this updated, check this and migrate existing old rules when version needs to be updated.

Reply:
Isn't it the responsibility of the schema updater to update the schema version, given that developers will be the ones modifying the models? I moved the constant to a config.py file, but I'm not sure if that's the best approach. What do you think?

Reply:
We will be responsible for updating the rules in our internal repository. It would be good idea to indicate which version they rules use and update the version when migrating to the new schema.

Reply:
Got it, I don't know how much will schema change in time, maybe we can keep different models for different versions if backward compatibility is not an option, while giving time to update rules to new versions, also that allow us to keep generating all schema versions for users to check an update or have different migration functions between versions to adapt them.